c/httpfileservice.c

```diff
@@ -452,6 +460,76 @@ void respondWithUnixFileMetadata(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void directoryChangeOwnerAndRespond(HttpResponse *response, char *path,
+        char *usrId, char *grpId, char *Recursive, char *pattern) {
+# define RETURN_MESSAGE_SIZE (PATH_MAX + 50)
+
+  char message[RETURN_MESSAGE_SIZE] = {0};
+  strncpy(message,"", RETURN_MESSAGE_SIZE);
+  UserInfo  userInfo = {0};
+  GroupInfo groupInfo = {0};
+
+  int returnCode = 0, reasonCode = 0;
+  int recursive = 0;
+  int userId, groupId, status;
+
+  if (!strcmp(strupcase(Recursive),"TRUE")) {
+    recursive = 1;
+  }
+
+  /* Evaluate user ID */
+  if (usrId == NULL) {
+    userId = -1;
+  }
+  else if (stringIsDigit(usrId)) {
+    userId = atoi(usrId);
+  }
+  else {
+    status = gidGetUserInfo(usrId, &userInfo, &returnCode, &reasonCode);
+    if (status == 0) {
+      userId = userInfoGetUserId ( &userInfo);
+    }
+    else {
+      sprintf(message, "Bad Input: User %s NOT found", usrId);
```

Comment:
If the `usrId` variable is large enough, you'll get a buffer overflow.

Reply:
Changed to snprintf.